### PR TITLE
Docs: Improve Gutenberg release documentation

### PR DESCRIPTION
**Pull request by anton-vlasenko**
## Description
We need to update Gutenberg release documentation to better explain if one needs to synchronize WordPress trunk and publish the npm packages or not.
This always causes some confusion during the release process.

## How has this been tested?
This PR doesn't have to be tested.

## Screenshots
No screenshots are needed.

## Types of changes
Improvement

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by anton-vlasenko** [comment]
I'm not sure whether we should include this sentence or not:
`However, if you are still unsure whether you should do it or not, please ask in the #core-editor Slack channel.`

**Comment by noisysocks** [comment]
It might be worth waiting to see what comes of this discussion: https://github.com/WordPress/gutenberg/discussions/37820

cc. @gziolo @ockham

**Comment by gziolo** [comment]
> It might be worth waiting to see what comes of this discussion: #37820

We can land the changes for now. We will update them again once we have something working.

**Comment by gziolo** [comment]
@luisherranz, I think you had a similar concern. Does the proposed revision look helpful to you?

File: docs/contributors/code/release.md [hunk 1]
```diff
@@ -160,7 +160,11 @@ There is also an option to perform [Standalone Bugfix Package Releases](#standal
 
 ### Synchronizing WordPress Trunk
 
-For each Gutenberg plugin release, WordPress trunk should be synchronized. Note that the WordPress `trunk` branch can be closed or in "feature-freeze" mode. Usually, this happens between the first `beta` and the first `RC` of the WordPress release cycle. During this period, the Gutenberg plugin releases should not be synchronized with WordPress Core.
+For each Gutenberg plugin release, WordPress trunk should be synchronized.
+
+Note that the WordPress `trunk` branch can be closed or in "feature-freeze" mode. Usually, feature freeze in WordPress Core happens about 2 weeks before Beta 1 and remains in effect until RC1 when the `trunk` gets branched. During this period, the Gutenberg plugin releases should not be synchronized with WordPress Core.
+
+A different person usually synchronizes the WordPress `trunk` branch and publishes the npm packages. So you shouldn't worry about it for the normal plugin release process. However, if you are still unsure whether you should do it or not, please ask in the #core-editor Slack channel.
```
Comment (getdave) [inline docs/contributors/code/release.md]:
```suggestion
A different person usually synchronizes the WordPress `trunk` branch and publishes the npm packages. Therefore, you typically shouldn't need to worry about handling this for the normal plugin release process. However, if you are still unsure, ask in [the #core-editor Slack channel](https://wordpress.slack.com/archives/C02QB2JS7).
```

**Review by getdave — APPROVED** [review]
Nit: Minor grammatical suggestions. Otherwise it's great to have additional clarity as this avoids unnecessary questions and confusing around the release process 👏

File: docs/contributors/code/release.md (hunk 1 above)
Comment (anton-vlasenko) [reply docs/contributors/code/release.md]: Thank you, @getdave !

**Comment by luisherranz** [comment]
> Does the proposed revision look helpful to you?

It does, yes 🙂

**Review by tellthemachines — COMMENTED** [review]
This is a good improvement, thanks Anton!

**Pull request merged**